Geta.EPi.Extensions/UrlExtensions.cs

```diff
@@ -75,5 +75,45 @@ private static Uri GetBaseUri()
                 ? SiteDefinition.Current.SiteUrl
                 : new Uri(HttpContext.Current.Request.Url.GetLeftPart(UriPartial.Authority));
         }
+
+        /// <summary>
+        ///     Gets the Url type of provided Url
+        /// </summary>
+        /// <param name="url"></param>
+        /// <returns>The type of Url (Internal, External). Returns Unknown if resolution fails.</returns>
+        public static UrlType GetUrlType(this Url url)
+        {
+            if (url.IsAbsoluteUri)
```

Comment:
If you add link to internal content in CMS, then yes - it is converted to relative Url. I really doubt that this logic is baked in Url class itself because it would be just too slow, but need to check. In this case you can not be sure where Url object came from as this is extension method that can be called on any Url instance.